server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java

```diff
@@ -234,7 +235,7 @@ public void parse(DocumentParserContext context) throws IOException {
                 context.addToFieldNames(fieldType().name());
             }
         } finally {
-            context.path().setWithinLeafObject(false);
+            context.path().setWithinLeafObject(isWithinLeaf);
```

Comment:
Nice catch

Reply:
Can you open a PR against main for this if it's a general bug? We don't need to fix it in this branch.

Reply:
https://github.com/elastic/elasticsearch/pull/118380